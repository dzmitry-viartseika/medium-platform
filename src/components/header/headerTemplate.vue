<template>
  <div class="app-header">
    <div class="app-header-logo">
      <div class="app-header-logo__text" click="goToRoute()">
        Conduit
      </div>
    </div>
    <div class="app-header-menu">
      <header-menu-template/>
    </div>
  </div>
</template>

<script>
import HeaderMenuTemplate from '@/components/header/headerMenuTemplate';

export default {
  name: 'HeaderTemplate',
  components: { HeaderMenuTemplate },
  computed: {
    checkHomeRoute() {
      return this.$route.name !== 'home';
    },
  },
  methods: {
    goToRoute() {
      if (this.checkHomeRoute) {
        this.$router.push('/');
      }
      return false;
    },
  },
};
</script>

<style scoped lang="scss">
  @import "../../assets/scss/variables";

  .app-header {
    display: flex;
    justify-content: space-between;
    padding: 20px 0;
    align-items: center;

    &-logo {

      &__text {
        cursor: pointer;
        font-size: 30px;
        color: $colorBtn;
      }
    }
  }
</style>
