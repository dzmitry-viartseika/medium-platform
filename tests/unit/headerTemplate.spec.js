import { shallowMount, createLocalVue } from '@vue/test-utils';
import headerTemplate from '@/components/header/headerTemplate.vue';
import Vuex from 'vuex';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(Vuex);
const store = new Vuex.Store({});
localVue.use(VueRouter);
const router = new VueRouter();

describe('headerTemplate.vue', () => {
  test('snapshot', () => {
    const $store = {
      state: {
        userInfo: 1,
      },
      getters: {
        userInfo: () => ({}),
      },
    };
    const wrapper = shallowMount(headerTemplate, {
      localVue,
      store,
      router,
      mocks: {
        $store,
      },
    });
    wrapper.setData({ userPhoto: true });
    expect(wrapper.find('.app-info-profile__photo')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.app-header-logo__text').text()).toBe('Conduit');
  });
});
