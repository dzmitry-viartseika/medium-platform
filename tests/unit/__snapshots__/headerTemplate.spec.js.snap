// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerTemplate.vue snapshot 1`] = `
<div class="app-header">
  <div class="app-header-logo">
    <div class="app-header-logo__text">
      Conduit
    </div>
  </div>
  <div class="app-header-menu">
    <header-menu-template-stub></header-menu-template-stub>
  </div>
</div>
`;
